bookwyrm/templates/shelf/shelf.html

```diff
@@ -124,6 +124,10 @@ <h2 class="title is-3">
                 {% endif %}
                 {% endwith %}
             </h2>
+            {% if books|length > 0 %}
```

Comment:
If the user filters a shelf with keywords that find nothing, this will remove the filter box, forcing them to refresh the page to see the unfiltered results.

You could fix this by doing something like this:

1. change the line here to `{% if books|length > 0 or is_filtered %}`
2. add a hidden input to the `search_filter_field.html` file: `<input hidden name="is_filtered" value="true">`
3. Add a value to the shelf View data: `"is_filtered": is_filtered = request.GET.get("is_filtered")`